### PR TITLE
KAA-738: Exceptions while stopping data collection demo. Fixed.

### PR DESCRIPTION
**Pull request by v-chyzhevskyi**
Fixed exceptions while stopping data collection demo.

### CODE REVIEW

File: client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/channel/impl/DefaultChannelManager.java [hunk 1]
```diff
@@ -537,7 +537,11 @@ public void run() {
                         channel.sync(task.getTypes());
                     }
                 } catch (InterruptedException e) {
-                    LOG.debug("[{}] Worker is interrupted", channel.getId(), e);
+                    if(stop){
```
Comment (ashvayka) [inline client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/channel/impl/DefaultChannelManager.java]: code formatting !

File: client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/channel/impl/DefaultChannelManager.java [hunk 2]
```diff
@@ -537,7 +537,11 @@ public void run() {
...
+                        LOG.debug("[{}] Worker is interrupted.", channel.getId());
+                    }else{
+                        LOG.warn("[{}] Worker is interrupted. Stack trace: ", channel.getId(), e);
```
Comment (ashvayka) [inline client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/channel/impl/DefaultChannelManager.java]: LOG.warn("[{}] Worker is interrupted.", channel.getId(), e);

File: client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/channel/impl/channels/DefaultOperationTcpChannel.java [hunk 3]
```diff
@@ -229,10 +229,10 @@ public void run() {
                     if (Thread.currentThread().isInterrupted()) {
                         if (channelState != State.SHUTDOWN) {
                             LOG.warn("Failed to read from the socket for channel [{}]. Stack trace: ", getId(), e);
-                            LOG.warn("Socket connection for channel [{}] was interrupted: ", e);
+                            LOG.warn("Socket connection for channel [{}] was interrupted: ", getId(), e);
                         } else {
-                            LOG.debug("Failed to read from the socket for channel [{}]. Stack trace: ", getId(), e);
-                            LOG.debug("Socket connection for channel [{}] was interrupted: ", e);
+                            LOG.debug("Failed to read from the socket for channel [{}].", getId());
```
Comment (ashvayka) [inline client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/channel/impl/channels/DefaultOperationTcpChannel.java]: you can remove this message.

File: client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/channel/impl/channels/DefaultOperationTcpChannel.java [hunk 4]
```diff
@@ -229,10 +229,10 @@ public void run() {
                     if (Thread.currentThread().isInterrupted()) {
                         if (channelState != State.SHUTDOWN) {
                             LOG.warn("Failed to read from the socket for channel [{}]. Stack trace: ", getId(), e);
```
Comment (ashvayka) [inline client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/channel/impl/channels/DefaultOperationTcpChannel.java]: you can remove this message.

**Comment by ashvayka** [comment]
LGTM

**Comment by Oldnickus** [comment]
retest this please

**Pull request merged**